staging/src/k8s.io/client-go/util/workqueue/queue.go

```diff
@@ -23,60 +23,64 @@ import (
 	"k8s.io/utils/clock"
 )
 
-type Interface interface {
-	Add(item interface{})
+type Interface TypedInterface[any]
+
+type TypedInterface[T comparable] interface {
+	Add(item T)
 	Len() int
-	Get() (item interface{}, shutdown bool)
-	Done(item interface{})
+	Get() (item T, shutdown bool)
+	Done(item T)
 	ShutDown()
 	ShutDownWithDrain()
 	ShuttingDown() bool
 }
 
 // Queue is the underlying storage for items. The functions below are always
 // called from the same goroutine.
-type Queue interface {
+type Queue[T comparable] interface {
```

Comment:
Why does this struct not have the `Typed` approach?  i.e.

```go
type Queue TypedQueue[any]

type TypedQueue[T comparable] interface {
```

Reply:
Because this got merged yesterday only and I went by the assumption that it is okay to break the interface, given that it has no users in k/k and is not in any released version: https://github.com/kubernetes/kubernetes/pull/123347